src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -4792,5 +4792,111 @@ .maxstack  2
 }
 ");
         }
+
+        [Theory]
+        [InlineData("class")]
+        [InlineData("struct")]
+        [WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1598252")]
+        public void StaticAndInstanceConstructors_01(string type)
+        {
+            string source = $$"""
+                using System;
+                {{type}} S
+                {
+                    static S()
+                    {
+                        Console.WriteLine("static constructor");
+                    }
+                    public S(int p)
+                    {
+                        Console.WriteLine("instance constructor: {0}", p);
+                    }
+                    public S() : this(0)
+                    {
+                    }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        _ = new S();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            VerifyExplicitlyDeclaredInstanceConstructors(
+                comp.GlobalNamespace.GetTypeMember("S"),
+                "S..ctor(System.Int32 p)",
+                "S..ctor()");
+            VerifyExplicitlyDeclaredInstanceConstructors(
+                ((Compilation)comp).GlobalNamespace.GetTypeMember("S"),
```

Comment:
What is the difference when obtaining the type symbol in this way, versus without the cast?

Reply:
The symbols from `Compilation` are public API `ISymbol`, while the symbols from `CSharpCompilation` are internal `Symbol`. Since the `ISymbol` instances are generated from the `Symbol` instances, this additional test may not be that useful, but it seemed simple to include.